online/src/main/scala/ai/chronon/online/FetcherBase.scala

```diff
@@ -49,43 +50,57 @@ class FetcherBase(kvStore: KVStore,
                   metaDataSet: String = ChrononMetadataKey,
                   timeoutMillis: Long = 10000,
                   debug: Boolean = false)
-    extends MetadataStore(kvStore, metaDataSet, timeoutMillis) {
+    extends MetadataStore(kvStore, metaDataSet, timeoutMillis)
+    with FetcherCache {
+  import FetcherBase._
 
-  private case class GroupByRequestMeta(
-      groupByServingInfoParsed: GroupByServingInfoParsed,
-      batchRequest: GetRequest,
-      streamingRequestOpt: Option[GetRequest],
-      endTs: Option[Long],
-      context: Metrics.Context
-  )
-
-  // a groupBy request is split into batchRequest and optionally a streamingRequest
-  // this method decodes bytes (of the appropriate avro schema) into chronon rows aggregates further if necessary
-  private def constructGroupByResponse(batchResponsesTry: Try[Seq[TimedValue]],
+  /**
+    * A groupBy request is split into batchRequest and optionally a streamingRequest. This method decodes bytes
+    * (of the appropriate avro schema) into chronon rows aggregates further if necessary.
+    *
+    * @param batchResponses a BatchResponses, which encapsulates either a response from kv store or a cached batch IR.
+    * @param streamingResponsesOpt a response from kv store, if the GroupBy was streaming data.
+    * @param oldServingInfo the GroupByServingInfo used to fetch the GroupBys.
+    * @param queryTimeMs the Request timestamp
+    * @param startTimeMs time when we started fetching the KV store
+    * @param overallLatency the time it took to get the values from the KV store
+    * @param context the Metrics.Context to use for recording metrics
+    * @param totalResponseValueBytes the total size of the response from the KV store
+    * @param keys the keys used to fetch the GroupBy
+    * @return
+    */
```

Comment:
I'm not sure about leaving these docs in. I might remove them

Reply:
I don't mind having these around

Reply:
I recommend dropping comments for every param - it adds a lot of noise to code and eventually hurts readability.

I think for a lot of these, the names are descriptive enough. 
In general comments should be either describing very high level control flow, or a specific non obvious behavior.

Reply:
@nikhilsimha I hear you on the added noise hurting readability. I removed all the `@param`s but left a few comments beside the parameters I think aren't fully descriptive from name alone. Let me know if it looks good to you! :)